nixos/modules/services/cluster/pacemaker/default.nix

```diff
@@ -43,7 +43,9 @@ in
     systemd.packages = [ cfg.package ];
     systemd.services.pacemaker = {
       wantedBy = [ "multi-user.target" ];
+      path = [ pkgs.coreutils pkgs.iproute2 pkgs.ocf-resource-agents ];
       serviceConfig = {
+        ExecStartPost = "${pkgs.coreutils}/bin/chown -R hacluster:pacemaker /var/lib/pacemaker";
```

Comment:
Why is this necessary? Shouldn’t the tmpfilesd rule above handle this?

Reply:
I can't recall been a year let me see if its still needed,could be detritus from testing.

Reply:
Ok so its not "technically" needed but what can happen is if the main daemon running as root saves the current configuration in the event of a separate node failure, the pengine backup could get saved as root and not pacemaker and causes pacemaker to die when it tries to start and read the prior configuration after fork() in one of the children processes.

I can't seem to get it to replicate though so maybe it was a bug in earlier versions? But basically that was the rationale its less systemd and more the interaction of the daemon and its children that run as the hacluster user. I could swap it to just be a find and to find any non user owned files in the state dir and chown only those this is a bit sledgehammery.